apis/duck/v1/source_types.go

```diff
@@ -81,6 +81,22 @@ type SourceStatus struct {
 	// Source.
 	// +optional
 	SinkURI *apis.URL `json:"sinkUri,omitempty"`
+
+	// CloudEventAttributes are the specific attributes that the Source uses
+	// as part of its CloudEvents.
+	// +optional
+	CloudEventAttributes []CloudEventAttributes `json:"ceAttributes,omitempty"`
+}
+
+// CloudEventAttributes specifies the attributes that a Source
+// uses as part of its CloudEvents.
+type CloudEventAttributes struct {
+
+	// Type refers to the CloudEvent type attribute.
+	Type string `json:"type,omitempty"`
+
+	// Source is the CloudEvents source attribute.
+	Source string `json:"source,omitempty"`
```

Comment:
Not sure we can reliably populate the `ce-source` attribute on a control plane component like this.

Source can be highly variable for a given event producer for a given `ce-type`.

Reply:
no, we can't. We already have a lot of variability in our current sources in eventing and eventing-contrib. That's why it's optional. Or it shouldn't even be here. If it's not, then we cannot do the refactor while maintaining the current EventType functionality... And I kind of wanted to have the refactor first, and then address EventType (I'm more inclined towards removing broker and source from it, although not entirely sure about the latter.)

Reply:
ok, just wanted to get Mikes input here, thanks Mike!

Reply:
talked w/ Nacho - we'll leave this in for now, but still have an issue open to revisit this in a larger conversation on discovery.

Reply:
sounds good, thanks for the input guys!